symplyphysics/laws/thermodynamics/compressibility_factor_via_intermolecular_force_potential.py

```diff
@@ -38,27 +39,32 @@
 intermolecular_force_potential = Function("intermolecular_force_potential", units.energy, real=True)
 temperature = Symbol("temperature", units.temperature, positive=True)
 
+infinite_distance = Quantity(S.Infinity, dimension=units.length)
+
 law = Eq(
     compressibility_factor,
     1 + 2 * pi * number_of_particles / volume
     * Integral(
         (1 - exp(-1 * intermolecular_force_potential(intermolecular_distance) / (units.boltzmann_constant * temperature))) * intermolecular_distance**2,
-        (intermolecular_distance, 0, S.Infinity),
+        (intermolecular_distance, 0, infinite_distance),
     )
 )
 
 # Calculate the compressibility factor for the model of hard spheres
 
-_sphere_radius = Symbol("sphere_radius", units.length, positive=True)
+_sphere_diameter = Symbol("sphere_diameter", units.length, positive=True)
 
-_hard_spheres_potential = Piecewise(
-    (S.Infinity, intermolecular_distance < _sphere_radius),
-    (0, intermolecular_distance >= _sphere_radius),
-)
+_hard_spheres_potential = hard_spheres_potential.law.rhs.subs({
+    hard_spheres_potential.distance: intermolecular_distance,
+    hard_spheres_potential.sphere_diameter: _sphere_diameter,
+    # Infinite quantity cannot be integrated properly - convert it to SymPy Infinity
+    # Note, that this way we lose information about its dimension
+    hard_spheres_potential.infinite_potential: hard_spheres_potential.infinite_potential.scale_factor,
```

Comment:
Plus I don't think we should do extra work taking out infinity from the scale factor every time it appears in expressions. `sympy` will simplify everything on its own if we do not put infinity inside a quantity.